### DISCUSSION TITLE
Cartpushers stuck

### DISCUSSION THREAD

**Discussion by catalin-duta** [Keriew/augustus#663] [Q&A]
I have one cartpusher stuck for many years here (mid-map-ish)
![Screenshot_20220831-212739723](https://user-images.githubusercontent.com/93785920/187753249-16120c4c-b56b-4cac-8226-94f5ccbc9827.jpg)
Not sure what happens here. I thought that cartpushers were supposed to recalculate their destination every once in a while.

Then there are more recent ones here (North entrance):
![Screenshot_20220831-212752800](https://user-images.githubusercontent.com/93785920/187753492-a4be8619-29dc-432d-8bfa-085073431ba8.jpg)
They got stuck most likely after the granaries were destroyed by a random Caesar attack (really no idea why).

Save attached- build 3.1.0.514
[8Valentia1q.zip](https://github.com/Keriew/augustus/files/9464039/8Valentia1q.zip)

**Comment by crudelios** [comment]
There has been a cartpusher fix after 514, can you check please?

**Reply by catalin-duta** [reply]
Reloaded the save in 3.1.0.528 (latest build): the 7 cartpushers are joined by another one, after a few months, but they still don't move.
[8Valentia1q2.zip](https://github.com/Keriew/augustus/files/9464622/8Valentia1q2.zip)
Clearing the rubble seems to get them moving but not the one by the reservoir (technically on grass).

There also seems to be something off in the recent builds with:
- buildings taking fire or collapsing with no notice
- pestilence warnings popping up with no location, that I can't find after looking at all the houses
- houses with sickness randomly bursting into flames
- at the end of it, the rubble ends up saying Nowhere instead of the building name
- I've also seen cartpushers say they are going to Nowhere

**Reply by Keriew** [reply]
> buildings taking fire or collapsing with no notice

Fixed that one, I overlooked how a function worked. Looking into the others.

**Reply by Ouaz** [reply]
> houses with sickness randomly bursting into flames

If the "Pestilence" video was played, it's the "regular" plague, and at least one random house is burned.
If you got the "Plague in the city" message, one of your house reaches 100 sickness (red column on the overlay), then there's two outcomes:
- The house burns if the global city health is < 40 (like the regular plague)
- If city health is > 40, the house is only quarantined (some dwellers die, and nobody can move in), until a doctor or surgeon comes to cure it by fumigation.

> pestilence warnings popping up with no location, that I can't find after looking at all the houses

Yeah, that's the normal behaviour, as there can be multiple locations. Maybe the "Go to location" button will be added, to show at least one of the impacted areas.

**Reply by catalin-duta** [reply]
> > houses with sickness randomly bursting into flames
> 
> If the "Pestilence" video was played, it's the "regular" plague, and at least one random house is burned.

The message does not include a video (at least not on Android). Or is it supposed to be at some point in the past?
And there's also that: I have some lower-quality housing, with higher sickness. It's not those ones that burn into flames, it's random houses, halfway across the city, within 5 squares of a clinic.
 
> > pestilence warnings popping up with no location, that I can't find after looking at all the houses
> 
> Yeah, that's the normal behaviour, as there can be multiple locations. Maybe the "Go to location" button will be added, to show at least one of the impacted areas.

But is it sure there *is* a house? ? Most of the time I can't seem to find it, at all. Is there no way to tone down this random pestilence?

**Reply by Ouaz** [reply]
> The message does not include a video (at least not on Android). Or is it supposed to be at some point in the past?
And there's also that: I have some lower-quality housing, with higher sickness. It's not those ones that burn into flames, it's random houses, halfway across the city, within 5 squares of a clinic.

Mmmh, this has been changed recently. I tested it on PC and it worked, it should work on Android too. If it's random houses that burn into flames, the "Pestilence" video should be definitely played (the house burns not because of sickness, but because of the global city health (what says the health advisor). This is the C3 original plague system.

Can you post a save before the random pestilence starts to happen?

**Reply by catalin-duta** [reply]
Here is one example, build 3.1.0.528 I think.
![Screenshot_20220901-225049759](https://user-images.githubusercontent.com/93785920/188001390-f572ce25-ef22-456d-99e8-5e5247448bb9.jpg)
This is the message I get (no video).
[8Valentia1r.zip](https://github.com/Keriew/augustus/files/9472975/8Valentia1r.zip)
Fast forwarding time, I get this several times, for different houses. And I never see what gets fumigated (💀), if anything.
I'm not getting flames now, not sure why.

**Reply by Ouaz** [reply]
Screenshot:
<details>
  <summary>Details</summary>
<img src="https://user-images.githubusercontent.com/11946570/188009714-eff402fb-28f5-491d-8995-cb0f962a0c34.png">
</details>
These 3 low rank houses (a large shack and two tents) have a high sickness value and are plagued one after each other, because they don't have access to healthcare facilities, and get contaminated by infected cartpushers. The way health is calculated has changed, and sickness does not decrease anymore with doctor coverage.

You couldn't know as it is explained in the new manual which will come with the 3.2 release. Gamerzahk gave a sneak peek at it: https://www.youtube.com/watch?v=rCLeFdtHrww&t=1593s

However, you can still use feeder houses. Just place them in a way that they can't be infected by cartpushers passing by.
![safe](https://user-images.githubusercontent.com/11946570/188012883-64da33ae-b36c-45a2-a5a8-e6b57e6adaaf.png)

**Comment by catalin-duta** [comment]
One other small comment related to the second situation (rubble): that was caused by one of Caesar's armies.
- I've received many requests for goods, all fulfilled
- one request for army support i didn't manage in time
Many years after that, the favor went way down, then Caesar sent the army. They destroyed a few buildings, then changed their mind.

For 10 years (47BC-37BC) the favor was between 45-55.
This is some time after the failed army support.
Then in 27BC ( I don't find any saves in between) the favor is at 25. Without missing any request for goods. In the years after it keeps going down.

Is this how it's supposed to work, or maybe I am missing something? Is it a must that I send gifts?

**Reply by Keriew** [reply]
You lose 2 favor/year if your salary is equal to your rank. (So if you pay yourself a procurator salary as a procurator for example.)
You only lose 1 favor/year if you pay yourself below your rank, but you lose more if you pay yourself above the rank. Aside from not fulfilling the requests and the yearly favor drop, you may also lose favor from being in debt.

In the example it looks like you were paying yourself proper salary for ten years and it resulted in drop of 20 favor. That seems in line with what should happen. That means that yes, to keep Caesar happy indefinitely you have to keep sending him gifts after requests run out, but you can greatly reduce the need for them by dropping your salary.

**Comment by githubfan1** [comment]
I would suggest a neutral position: no drop if personal salary is zero keeping losing -2 and -1 favor points as explained above. Player would increase salary in case there would be a need to increase favor by sending a gift and then back to zero salary. Personally rarely I send gifts to caesar. I think that playing with Augustus is much easier to accomplish all the requirements to win any scenario. I already played all peaceful campaigns without sending gifts and now I am playing dangerous cities (Tingis so far) no need to gift him/her.
Thanks.
